test/utils/setup.ts

```diff
@@ -0,0 +1,150 @@
+import { Span } from '@opentelemetry/api';
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-grpc';
+import { registerInstrumentations } from '@opentelemetry/instrumentation';
+import { detectResources, envDetector } from '@opentelemetry/resources';
+import { SimpleSpanProcessor } from '@opentelemetry/sdk-trace-base';
+import { NodeTracerProvider } from '@opentelemetry/sdk-trace-node';
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
+import realFetch from 'cross-fetch';
+import dotenv from 'dotenv';
+import { join } from 'path';
+import { File, Suite, TestContext, vi } from 'vitest';
+import { BaseClient, CacheImpl, XataApiClient } from '../../packages/client/src';
+import { getHostUrl, HostProvider, isHostProviderAlias } from '../../packages/client/src/api/providers';
+import { XataClient } from '../../packages/codegen/example/xata';
+import { buildTraceFunction } from '../../packages/plugin-client-opentelemetry';
+import { teamColumns, userColumns } from '../mock_data';
+
+// Get environment variables before reading them
+dotenv.config({ path: join(process.cwd(), '.env') });
+
+const apiKey = process.env.XATA_API_KEY ?? '';
+if (apiKey === '') throw new Error('XATA_API_KEY environment variable is not set');
+
+const workspace = process.env.XATA_WORKSPACE ?? '';
+if (workspace === '') throw new Error('XATA_WORKSPACE environment variable is not set');
+
+const host = getProvider(process.env.XATA_API_PROVIDER);
+const fetch = vi.fn(realFetch);
+
+export type EnvironmentOptions = {
+  cache?: CacheImpl;
+};
+
+export type TestEnvironmentResult = {
+  api: XataApiClient;
+  client: XataClient;
+  baseClient: BaseClient;
+  database: string;
+  workspace: string;
+  clientOptions: {
+    databaseURL: string;
+    fetch: typeof fetch;
+    apiKey: string;
+    branch: string;
+    cache?: CacheImpl;
+  };
+  hooks: {
+    beforeAll: (ctx: Suite | File) => Promise<void>;
+    afterAll: (ctx: Suite | File) => Promise<void>;
+    beforeEach: (ctx: TestContext) => Promise<void>;
+    afterEach: (ctx: TestContext) => Promise<void>;
+  };
+};
+
+export async function setUpTestEnvironment(
+  prefix: string,
+  { cache }: EnvironmentOptions = {}
+): Promise<TestEnvironmentResult> {
+  const { trace, tracer } = await setupTracing();
+
+  const id = Math.round(Math.random() * 100000);
```

Comment:
I'm not sure, but do we want to use a uuid for this?

Reply:
Yeah, we faced some random 422 already. I was thinking of timestamp instead